src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,160 @@
+package duke;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+/** Deals with user interactions. */
+public class Ui {
+    public static final String HORIZONTAL_BAR =
+            "____________________________________________________________";
+    public static final String NEWLINE = System.lineSeparator();
+    public static final String INDENTATION = "    ";
+
+    /**
+     * Accepts user input from stdin.
+     *
+     * @param inputReader used to read user input
+     * @return String representing a user command
+     * @throws IOException when BufferedReader fails to read user input
+     */
+    protected String readCommand(BufferedReader inputReader) throws IOException {
+        String command;
+        command = inputReader.readLine().trim();
+        return command;
+    }
+
+    /**
+     * Prints given text with indentation: specified in duke.Duke class.
+     *
+     * @param text used for formatting and printing
+     */
+    protected void printTextWithIndentation(String text) {
+        System.out.println(INDENTATION + text);
+    }
+
+    /**
+     * Prints greeting message.
+     */
+    protected void greet() {
+        String logo = "  __  __        _____       _           _   " + NEWLINE
+                + " |  \\/  |      |  __ \\     | |         | |  " + NEWLINE
+                + " | \\  / |_ __  | |__) |___ | |__   ___ | |_ " + NEWLINE
+                + " | |\\/| | '__| |  _  // _ \\| '_ \\ / _ \\| __|" + NEWLINE
+                + " | |  | | |    | | \\ \\ (_) | |_) | (_) | |_ " + NEWLINE
+                + " |_|  |_|_|    |_|  \\_\\___/|_.__/ \\___/ \\__|";
+        System.out.println(logo);
+        printTextWithIndentation(HORIZONTAL_BAR);
```

Comment:
Maybe it would be cleaner if you created another helper method printParagraphWithIndentation that you can call on a paragraph/string[] of lines so that you don't have to call printTextWithIndentation multiple times?